geode-core/src/main/java/org/apache/geode/net/SSLParameterExtension.java

```diff
@@ -29,9 +29,9 @@
   /**
    * Initialize the SSLParameterExtension.
    *
-   * @param config the DistributionConfig
+   * @param properties the Properties
    */
-  default void init(DistributionConfig config) {}
+  default void init(Properties properties) {}
```

Comment:
I would like to see some javadoc comments explaining both
1) why a breaking change was justified, and
2) documentation of property names and suitable values for each that need to be in this Properties object

Reply:
>why a breaking change was justified

Think of this change as taking something that was internal and making it public API. The original introduction of this API was intended to be public but included components that are internal thus making it effectively an internal API.

Reply:
>documentation of property names and suitable values for each that need to be in this Properties object

@mivanac If you can name these then I think it justifies the request I had to have a concrete domain object for this API rather than a wide open property bag.

Reply:
As described in https://cwiki.apache.org/confluence/display/GEODE/Introduction+of+SSL+Parameter+Extension,
idea was to implement generic solution for modification of SSL parameters. If the user wishes to manipulate SSL Parameters they would need to define class which implements new interface.
As such, Properties was used as generic input argument, and user will define which parameter he is interested in. For our case, parameter of interest is DISTRIBUTED_SYSTEM_ID.

Reply:
The system properties isn't really a good API though. I really think we should make a concrete class, even if over time we have to evolve it to have more information. It's controlled and strongly typed.

Reply:
@pivotal-jbarrett , you are proposing that `Properties` be replaced with a concrete class, wrapping a map? You did not request a change to the PR, so I'm otherwise ready to merge this as-is, and get it onto the 1.13 branch before it goes out, too.

Reply:
I’m not sure what the consensus is reading the threads. I see myself and a +1 in favor of something more strongly defined. I see the original PR for just properties with an approval by the same person that +1 the suggested change. *shrug*

Reply:
I suggest we leave the argument type as `Properties` but mark it `@Experimental` and get it into 1.13.0 in that state for now

Reply:
Then why change it at all for 1.13. It’s broken now. It will be broken later if we add @experimental knowing we will change it in the next release. Might as well save the trouble and skip this PR for 1.13 entirely and do it right in 1.14.

Reply:
Using an internal class in a public API is "more broken", and we want to address that immediately.  It remains unclear what a long-term solution should look like.  Properties may be distasteful to some but seems sufficient to fix the internal API leakage.  Adding @Experimental now allows us to refine the API in the future and hopefully deprecate the use of Properties someday.

Reply:
@pivotal-jbarrett can you share a code snippet of what you think a concrete class might look like?  I don't have the domain expertise to understand how restricting the ssl properties now might bite us in the future if additional properties become needed.